### PR TITLE
[#10569] missing loading icons for Instructor Sessions Page

### PR DESCRIPTION
**Pull request by jfrairigh**
Part of #10569 

Added boolean values to turn the spinners on when action is processing and off when action has finished. This code has fixed the remaining bullet points of issue #10569 for the Instructor Session Page. Spinners are still needed for buttons on the Instructor Session Edit Page.

### CODE REVIEW

**Comment by ChooJeremy** [comment]
Hi, can you fix the E2E test in the CI? https://travis-ci.org/github/TEAMMATES/teammates/builds/742791201?utm_source=github_status&utm_medium=notification

**Comment by jfrairigh** [comment]
@ChooJeremy I am looking it to this now. I'm new to E2E test so I'm reading through the documentation and CI logs to figure out what to do. It may take some time, but I'm sure I can figure it out.

**Comment by jfrairigh** [comment]
I have looked into this quite a bit and I running into an issue I'm not sure how to resolve. At the beginning of October I setup my local environment for Teammates. Then, I ran the front end, unit, and E2E test. All tests passed. Now I have ran the E2E tests against my local master branch ,which is a clean clone of teammates master repo and the tests fail. I have ran it multiple times and there are multiple tests that fail each time. Sometimes its as many 50 and sometimes its as few as 5. If the test were only giving me trouble in my working branch I would suspect it was the code I introduced, but since its happening in the clean master I'm not sure why the test are failing. Does anyone have any insight that may help me overcome this obstacle?

**Comment by ChooJeremy** [comment]
Hi @jfrairigh, what errors are you getting?

Sometimes an E2E test may fail due to the error `org.gradle.internal.serialize.PlaceholderException: Timeout loading page after 10000ms`. This is not unusual. However, we do run the failing tests multiple times (in the case of the CI, you'll notice that the tests was ran 5 times:

`39 tests completed, 3 failed`
> Task :e2eTestTry2
`3 tests completed, 2 failed`
> Task :e2eTestTry3
`2 tests completed, 1 failed`

**Comment by jfrairigh** [comment]
@ChooJeremy I am not questioning the need to fix the E2E test in the CI. https://travis-ci.org/github/TEAMMATES/teammates/builds/742791201?utm_source=github_status&utm_medium=notification
I'm trying to understand why the tests are failing on my local master branch that is a clean clone of upstream/master. I believe this is an indication of an environment configuration issue. However, I'm not sure what it could be. I first started working on Teammates' project in October. I followed the set-up instruction I then ran all tests (front-end, E2E, and unit tests) to verify my environment was set up correctly. At that time all test were passing. Now the E2E tests are not passing. I am wondering if anyone has encountered something like this before, and if so what they did to resolve it. I have looked through the Teammates' documentation and search forums as well. I am out of ideas. I have even recruited my husband who much more experienced than I am to help, but he is out of ideas as well. I believe I should fix the issue in my local environment before fixing the failed test dected by the CI. Here is a screenshot of my last run of the tests on my clean master branch:

image

**Comment by wkurniawan07** [comment]
Hi @jfrairigh, the E2E tests have indeed undergone some major improvements since October. You're most probably right in that it's an environment configuration issue; you can create a fresh copy of `src/e2e/resources/test.properties` by duplicating the contents of `src/e2e/resources/test.template.properties` and modifying the relevant values. This should get rid of the failures, except possibly in `InstructorFeedbackSessionsPageE2ETest` (which, according to Travis CI, is the one that needs fixing) and `TimezoneSyncerTest` (known problem, #9645).

**Comment by jfrairigh** [comment]
Thank you for the advise @wkurniawan07 that got rid of the failures except for the TimezoneSyncerTest (known problem, #9645). 
e2eTestOnEditedBranch

However, when I run it locally in the branch with the changes, the InstructorFeedbackSessionsPageE2ETest passes. Do you think is due to an environment configuration as well? I'm trying to reproduce the CI's results locally before attempting  to fix the test issue.

This step is a little harder than I anticipated. Thank you again for your help so far.

**Comment by jfrairigh** [comment]
I have finally been able to reproduce the same results from the E2E tests as the CI. Yeah! 
I will now look into fixing the E2E test.

**Comment by jfrairigh** [comment]
Changes have been made and it's ready for review.

**Comment by jfrairigh** [comment]
I noticed there is an issue created , #10682, to add spinners for copying operations. In that issue it asks for the spinners to be added to the buttons. In the code for this pull request I added the spinner to the page not the button.  Should I change the code to be consistent with style requested in #10682? Also, I did this on PR #10813, which has already been accepted and merged. I am planning to finish the work for issue #10569, and would like to know before working on it further.

**Comment by niqiukun** [comment]
> Should I change the code to be consistent with style requested in #10682?

Any advice from team leads? @madanalogy @rrtheonlyone

**Comment by madanalogy** [comment]
Issue #10682 does not request that a spinner be placed next to the button as far as I'm aware. It simply states that some kind of progress indication should be present

**Comment by jfrairigh** [comment]
@niqiukun Is there anything I need to do on my end to get progress this?

**Review by niqiukun — APPROVED** [review]
LGTM, thanks for the reminder!

File: src/e2e/resources/test_original.properties [hunk 1]
```diff
@@ -0,0 +1,107 @@
+#-----------------------------------------------------------------------------
+# This file contains some configuration values used during testing.
+# It should be placed in src\e2e\resources
+# Use '\' to escape ':' e.g., http\://google.com
+#-----------------------------------------------------------------------------
```
Comment (madanalogy) [inline src/e2e/resources/test_original.properties:5]: I believe this file is not supposed to be part of your PR. Please remove it and we can go ahead with merging your changes

**Review by madanalogy — CHANGES_REQUESTED** [review]

File: src/e2e/resources/test_original.properties (hunk 1 above)
Comment (jfrairigh) [reply src/e2e/resources/test_original.properties]: @madanalogy Oops! Sorry about that. I just deleted it from the commit.

**Review by madanalogy — APPROVED** [review]
LGTM! Thanks for your contribution!

**Pull request merged**